src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsJsonEncoding.cs

```diff
@@ -356,12 +356,22 @@ LaunchProfile ReadProfile(string name)
             ReadObject(property =>
             {
                 builder ??= ImmutableArray.CreateBuilder<(string Name, string Value)>();
-                builder.Add((property, ReadString()));
+                builder.Add((property, ReadEnvironmentVariableString()));
             });
 
             return builder?.ToImmutable() ?? ImmutableArray<(string Name, string Value)>.Empty;
         }
 
+        string ReadEnvironmentVariableString()
+        {
+            if (!reader.Read())
+            {
+                throw new JsonReaderException($"Cannot read string at {reader.LineNumber}.");
+            }
+
+            return reader.Value!.ToString();
```

Comment:
This will fail if the value is null in the JSON file. You cannot suppress this here. The only reason we can suppress it in the other cases is because we check `TokenType`, which isn't happening here.

Reply:
You're right. He'd have to check in his guard clause that `reader.TokenType != JsonToken.Null`.